### PR TITLE
beltagy2020longformer Summary

### PR DESCRIPTION
**Pull request by liaojh1998**
To be done by later today. I'm still reading through the paper.

### CODE REVIEW

**Comment by liaojh1998** [comment]
Hey @kraja99, sorry about the delay on the draft. I finally completed it so you can review it now. Thank you!

File: _summaries/beltagy2020longformer_2.md [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+
+    <p align="center">
+        <img src="beltagy2020longformer/2d.png" width="15%" height="15%">
+    </p>
+
+    In the case of a QA task, global attention is provided on all question tokens. Because the number of tokens with global attention is small relative to sequence length, the combined local and global attention is still bounded by $$\mathcal{O}(n)$$.
+
+    In implementation, there's a linear projection for sliding window attention and a separate linear projection for global attention per token. This method avoids the global and sliding window attentions from convoluting each other.
+
+    **Implementation Detail**: Modern PyTorch/Tensorflow libraries do not support the form of banded matrix multiplication required for the sliding window attention, so the authors implemented 3 ways of doing it: `loop`, `chunks`, and `cuda`. `loop` is used for testing, `chunks` is used for pre-training/fine-tuning, `cuda` is a highly optimized custom CUDA kernel that is used for the language modeling experiments.
```
Comment (kraja99) [inline _summaries/beltagy2020longformer_2.md:58]: Might be worthwhile to mention that different Q,K,V projections are used for computing local (sliding or dilated) versus global attention

File: _summaries/beltagy2020longformer_2.md [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+    <p align="center">
+        <img src="https://d3i71xaburhd42.cloudfront.net/925ad2897d1b5decbea320d07e99afa9110e09b2/8-Table8-1.png" width="50%" height="50%">
+        <br>
+        <img src="beltagy2020longformer/t8.png" width="30%" height="30%">
+    </p>
+
+    * **LED**
+        * Dataset: arXiv summarization dataset ([Cohan et al., 2018](https://aclanthology.org/N18-2097/)), which has 14.5K tokens at the 90th percentile of document lengths.
+        * Evaluation: ROUGE-1 (R-1), ROUGE-2 (R-2), and ROUGE-L (R-L) ([Lin, 2004](https://aclanthology.org/W04-1013/)).
+        * Results: State-of-the-art. Outperforms BigBird ([Zaheer et al., 2020](https://papers.nips.cc/paper/2020/hash/c8512d142a2d849725f31a9a7a361ab9-Abstract.html)).
```
Comment (kraja99) [inline _summaries/beltagy2020longformer_2.md:137]: Consider discussing the caveat that while performance with BigBird is close, BigBird is built off and continues training from Pegasus which is trained and pre-trained specifically for summarization.

File: _summaries/beltagy2020longformer_2.md [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+
+    <p align="center">
+        <img src="beltagy2020longformer/t11.png" width="30%" height="30%">
+    </p>
+
+## TL;DR
```
Comment (kraja99) [inline _summaries/beltagy2020longformer_2.md:143]: Great summary. You provide an in-depth look at the technical details of Longformer as well as present the results in an easily digestible manner.

**Review by kraja99 — COMMENTED** [review]
Great summary, added comments in the PR

**Comment by liaojh1998** [comment]
Thanks for the review Kiran. I added the following:
* Clarification on different linear projections used for global and sliding attention.
* Caveat that Longformer slightly outperforms BigBird, which is a model that starts from a pre-trained Pegasus, and is specifically designed and pre-trained for summarization.

**Review by zhaoyue-zephyrus — APPROVED** [review]

**Pull request merged**